packages/vue/src/components/molecules/SfDevice/SfDevice.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+  <div class="sf-device">
+    <div class="sf-device__wrapper">
+      <div class="sf-device__morph" @click="clicked">
+        <div class="sf-device__screen">
+          <slot />
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+const devices = ["mobile", "tablet", "laptop", "detect"];
+
+export default {
+  name: "SfDevice",
+  props: {
+    device: {
+      type: String,
+      default: "detect",
+      validator: (propValue) => {
+        return devices.includes(propValue);
+      },
+    },
+    switchOnClick: {
+      type: Boolean,
+      default: false,
+    },
+    switchInterval: {
+      type: Number,
+      default: 0,
+    },
+  },
+  data: function () {
+    return {
+      i: 0,
+      morphEl: null,
+      intervalID: -1,
+      devices: devices,
+      classes: ["sf-device__mobile", "sf-device__tablet", "sf-device__laptop"],
+    };
+  },
+  mounted() {
+    this.morphEl = this.$el.querySelector(".sf-device__morph");
+
+    if (this.device === "detect") {
+      if (typeof window.matchMedia !== "undefined") {
+        const isMobile = window.matchMedia(
```

Comment:
suggestion: window property doesn't work with SSR - can you try to use CSS for conditional rendering? It's the best practice.